gc/base/standard/ParallelGlobalGC.cpp

```diff
@@ -1292,6 +1329,25 @@ MM_ParallelGlobalGC::fixHeapForWalk(MM_EnvironmentBase *env, UDATA walkFlags, ui
 	return fixedObjectCount;
 }
 
+uintptr_t
+MM_ParallelGlobalGC::clearHeap(MM_EnvironmentBase *env,UDATA walkFlags, uintptr_t walkReason, MM_HeapWalkerObjectFunc walkFunction)
+{
+	memorySizes counter;
+	counter.freeBytes = 0;
+	counter.objectBytes = 0;
+	uintptr_t fixedObjectCount = 0;
+	OMRPORT_ACCESS_FROM_ENVIRONMENT(env);
+	omrtty_printf("ClearHeap CALLED.\n");
+	_heapWalker->allObjectsDo(env, walkFunction, &counter, walkFlags, false, false, true);
+	U_64 startTime = omrtime_hires_clock();
+	_extensions->globalGCStats.fixHeapForWalkTime = omrtime_hires_delta(startTime, omrtime_hires_clock(), OMRPORT_TIME_DELTA_IN_MICROSECONDS);
+	_extensions->globalGCStats.fixHeapForWalkReason = walkReason;
```

Comment:
I'll think  a bit if we want to affect globalGCStats.fixHeapForWalk* fields
This is not really a fixup walk, and more importanly we will overwrite the data from the pre-req fixup  that jsut happened before this clearing pass, and there is a posibility taht  VGC reporting for that fixup has not happened yet.

Reply:
This was copied from MM_ParallelGlobalGC::fixHeapForWalk(), I can definitely remove these lines.

Reply:
after somethinking let's leave it  but:
- we will put an assert that fixHeapForWalkReason is already FIXUP_DEBUG_TOOLING before overwritting it; note that reason is set to NONE at the start of each GC
- we will introduce another fixup reason (CLEAR_HEAP or potentially CLEAR_AND_FIXUP_HEAP that indirectly says we did 2 pases) to set it to
- optionally we can append the duration of this pass to the fixHeapForWalkTime (rather then overwritting it).